metrics/ci_integrations/test/source/buildkite/adapter/buildkite_source_client_adapter_test.dart

```diff
@@ -732,6 +742,235 @@ void main() {
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the build number equal to the fetched build's number",
+      () async {
+        const build = BuildkiteBuild(number: buildNumber);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.buildNumber, equals(buildNumber));
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the started at date equal to the fetched build's started at date if it is not null",
+      () async {
+        final build = BuildkiteBuild(startedAt: startedAt);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.startedAt, equals(startedAt));
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the started at date equal to the fetched build's finished at date if the build's started at date is null",
+      () async {
+        final build = BuildkiteBuild(startedAt: null, finishedAt: finishedAt);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.startedAt, equals(finishedAt));
+      },
+    );
+
+    test(
+      ".fetchOneBuild() returns a build data with the started at date equal to the current date time if the fetched build's started at and finished at dates are null",
+      () async {
+        const build = BuildkiteBuild(startedAt: null, finishedAt: null);
+        whenFetchOneBuild().thenSuccessWith(build);
+
+        final result = await adapter.fetchOneBuild(pipelineSlug, buildNumber);
+
+        expect(result.startedAt, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchOneBuild() maps fetched builds states according to the specification",
+      () async {
+        const states = [
+          BuildkiteBuildState.passed,
+          BuildkiteBuildState.failed,
+          BuildkiteBuildState.canceled,
+          BuildkiteBuildState.notRun,
+          BuildkiteBuildState.scheduled,
+          BuildkiteBuildState.blocked,
+          BuildkiteBuildState.notRun,
+          BuildkiteBuildState.finished,
+          BuildkiteBuildState.running,
+          BuildkiteBuildState.skipped,
+          null,
+        ];
+
+        const expectedStatuses = [
+          BuildStatus.successful,
+          BuildStatus.failed,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+          BuildStatus.unknown,
+        ];
+
+        final statesLength = min(states.length, expectedStatuses.length);
```

Comment:
This statement may cause the test to be unreliable. Consider the following case: 
1. Buildkite introduces a new state for builds. 
2. Someone changes this test case and adds a new state to the `states`, however, forgets to add the expected mapped value to the `expectedStatuses`.
3. Test passes but actually must fail.

Thus:
```suggestion
        final statesLength = states.length;
```
or 
```suggestion
        final statesLength = expectedStatuses.length;
```